sdk/search/Azure.Search.Documents/src/Indexes/Models/BlobExtractionMode.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Search.Documents.Indexes.Models
+{
+    /// <summary>
+    /// Defines which parts of a blob will be indexed by the blob storage indexer.
+    /// See <see href="https://docs.microsoft.com/azure/search/search-howto-indexing-azure-blob-storage" /> for details.
+    /// </summary>
+    public readonly partial struct BlobExtractionMode : IEquatable<BlobExtractionMode>
+    {
+        private readonly string _value;
+
+        /// <summary> Determines if two <see cref="BlobExtractionMode"/> values are the same. </summary>
+        public BlobExtractionMode(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string StorageMetadataValue = "storageMetadata";
+        private const string AllMetadataValue = "allMetadata";
+        private const string ContentAndMetadataValue = "contentAndMetadata";
+
+        /// <summary>
+        /// Specifies that only the standard blob properties and user-specified metadata will be indexed.
+        /// See <see href="https://docs.microsoft.com/azure/storage/storage-properties-metadata" /> for details.
+        /// </summary>
+        public static BlobExtractionMode StorageMetadata { get; } = new BlobExtractionMode(StorageMetadataValue);
+
+        /// <summary>
+        /// Specifies that storage metadata and the content-type specific metadata extracted from the blob content will be indexed.
+        /// See <see href="https://docs.microsoft.com/azure/search/search-howto-indexing-azure-blob-storage#content-type-specific-metadata-properties" /> for details.
+        /// </summary>
+        public static BlobExtractionMode AllMetadata { get; } = new BlobExtractionMode(AllMetadataValue);
```

Comment:
This should be `MetadataAndProperties`

Reply:
Why "MetadataAndProperties"?  If we change here, I want it changed in portal as well.

Reply:
Storage makes a distinction between [metadata](https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob-metadata) and [properties](https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob-properties). 
 Metadata are custom key/value pairs you can throw on a blob.  Properties are well known HTTP headers like Content-Type.  (FWIW, this is a newer distinction in the last couple of years.)  As somebody familiar with Storage, I'm not sure what `AllMetadata` means here.  Does it include both Properties and Metadata?

Reply:
For the sample, I'm keeping the names as they show in the portal. When we code these into the swagger, we should decide example what they should specify (not breaking existing customers) and how they should display.